testcharms/charm-hub/charms/juju-qa-test/charmcraft.yaml

```
@@ -1,4 +1,12 @@
 type: charm
-charmhub:
-    api_url: https://api.charmhub.io
-    storage_url: https://storage.snapcraftcontent.com
+bases:
+    - build-on:
+        - name: "ubuntu"
+          channel: "20.04"
+        - name: "ubuntu"
+          channel: "22.04"
+      run-on: 
```

```suggestion
      run-on: 
      - name: "ubuntu"
        channel: "18.04"
```

That gives us a charm that still runs on bionic for 2.9 testing.